pom.xml

```
@@ -109,6 +109,9 @@ flexible messaging model and an intuitive client API.</description>
     <testRetryCount>1</testRetryCount>
     <testJacocoAgentArgument/>
     <testHeapDumpPath>/tmp</testHeapDumpPath>
+    <!-- This property is used to generate reproducible builds. We do not use it, but we do not want to inherit
+    this time property from dependencies. By setting it to null, we get "now" as the build time. -->
+    <project.build.outputTimestamp></project.build.outputTimestamp>
```

```suggestion
    <project.build.outputTimestamp>10</project.build.outputTimestamp>
```

https://github.com/tisonkun/pulsar/pull/21 It seems this change can make the SQL tests passed now.

The config is copied from https://maven.apache.org/guides/mini/guide-reproducible-builds.html.

I'm trying locally to see if the timestamp is correctly resolved.

---

It seems force the timestamp to a past time is less than awesome, although the property can be updated on release.

Another possible cause is that we don't update the property in `buildtools` also, as well as we have some plugins that conflict with reproducible build (detected by `mvn artifact:check-buildplan `.

I'll debug on https://github.com/tisonkun/pulsar/pull/21 to find a good way while SQL tests keep failing on this patch.

---

Thank you for debugging this! I should have looked closer.